### PR TITLE
c#/10.prompt-validations

### PR DESCRIPTION
**Pull request by JasonSowers**
- [x] Is numbered and named correctly and following the spec
--
- [x] All files have been spell checked
- [x] All style cop/ linting issues have been addressed
- [x] Include all DCR changes
- [x] Explicit state save
- [x] Method name changes begin, end etc
- [x] Does not use mainDialog unless that file actually extends Dialog/ ComponentDialog
- [x] Compiles with latest packages
- [x] Runs per instructions in README
- [x] Add commented out code to use blob storage
- [x] Update to look for production/ local (bot configuration)
- [x] onTurnError at adapter
- [x] Include .env or appsettings.json or web.config with botFilePath and botFileSecret
- [x] Deployment scripts folder has been cleaned up and only includes msbot export files
- [x] README to include
- [x] msbot clone steps
- [x] Tools install/ uninstall
- [x] Concise set of steps to run the sample
- [x] C# sample -- Include .bot file in the build output
- [ ] All samples include the latest botfile (no "secretKey" field; replaced with "padLock"). Use latest MSBOT off the tools MyGet feed to generate the bot file
- [x] remove and sort usings
- [x] remove azuredeploy.* under deploymentscripts folder (if any)
- [x] Welcoming users is consistent & correct. See 13.basic bot for nodejs & 3. welcome user for C#

### CODE REVIEW

File: samples/csharp_dotnetcore/10.prompt-validations/Startup.cs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                    ConversationDialogState = conversationState.CreateProperty<DialogState>("DialogState"),
+                };
+
+                return accessors;
+            });
+        }
+
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory)
+        {
+            this._isProduction = env.IsProduction();
```
Comment (daveta) [inline samples/csharp_dotnetcore/10.prompt-validations/Startup.cs]: >env.IsProduction() [](start = 33, length = 18)

Move to top of constructor..

File: samples/csharp_dotnetcore/10.prompt-validations/readme.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+## Update packages
+- In Visual Studio right click on the solution and select "Restore NuGet Packages".
+  **Note:** this sample requires `Microsoft.Bot.Builder.Dialogs` and `Microsoft.Bot.Builder.Integration.AspNet.Core`.
+## Testing the bot using Bot Framework Emulator
+[Microsoft Bot Framework Emulator](https://github.com/microsoft/botframework-emulator) is a desktop application that allows bot 
+developers to test and debug their bots on localhost or running remotely through a tunnel.
+- Install the Bot Framework emulator from [here](https://aka.ms/botframeworkemulator).
+## Connect to bot using Bot Framework Emulator V4
+- Launch the Bot Framework Emulator.
+- File -> Open bot and navigate to `BotBuilder-Samples\samples\csharp_dotnetcore\04.simple-prompt` folder.
```
Comment (daveta) [inline samples/csharp_dotnetcore/10.prompt-validations/readme.md]: >04.simple-promp [](start = 81, length = 15)

Change to 10.prompt-validations

File: samples/csharp_dotnetcore/10.prompt-validations/PromptValidationsBot.cs [hunk 3]
```diff
@@ -36,9 +38,9 @@ public class PromptValidationsBot : IBot
         /// <param name="accessors">The state accessors this instance will be needing at runtime.</param>
         public PromptValidationsBot(BotAccessors accessors)
         {
-            _accessors = accessors ?? throw new ArgumentNullException(nameof(accessors));
-            _dialogs = new DialogSet(accessors.ConversationDialogState);
-            _dialogs.Add(new TextPrompt("name", CustomPromptValidatorAsync));
+            this._accessors = accessors ?? throw new ArgumentNullException(nameof(accessors));
```
Comment (daveta) [inline samples/csharp_dotnetcore/10.prompt-validations/PromptValidationsBot.cs]: >this [](start = 12, length = 4)

Hmm - against our style guidelines..

File: samples/csharp_dotnetcore/10.prompt-validations/PromptValidationsBot.cs [hunk 4]
```diff
@@ -56,44 +58,51 @@ public async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancel
...
+                await SendWelcomeMessageAsync(turnContext, cancellationToken);
+            }
+            else
+            {
+                await turnContext.SendActivityAsync($"{turnContext.Activity.Type} event detected", cancellationToken: cancellationToken);
             }
 
             // Save the new turn count into the conversation state.
-            await _accessors.ConversationState.SaveChangesAsync(turnContext, false, cancellationToken);
+            await this._accessors.ConversationState.SaveChangesAsync(turnContext, false, cancellationToken);
```
Comment (daveta) [inline samples/csharp_dotnetcore/10.prompt-validations/PromptValidationsBot.cs]: >this [](start = 18, length = 4)

this scattered about..

File: samples/csharp_dotnetcore/10.prompt-validations/PromptValidationsBot.cs [hunk 5]
```diff
@@ -56,44 +58,51 @@ public async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancel
...
 
             // We are only interested in Message Activities.
-            if (turnContext.Activity.Type != ActivityTypes.Message)
+            if (turnContext.Activity.Type == ActivityTypes.Message)
             {
-                return;
-            }
+                // Run the DialogSet - let the framework identify the current state of the dialog from
+                // the dialog stack and figure out what (if any) is the active dialog.
+                var dialogContext = await this._dialogs.CreateContextAsync(turnContext, cancellationToken);
```
Comment (daveta) [inline samples/csharp_dotnetcore/10.prompt-validations/PromptValidationsBot.cs]: >this [](start = 42, length = 4)

Style

**Comment by JasonSowers** [comment]
@daveta I have addressed all comments.

**Comment by JasonSowers** [comment]
@daveta done

**Pull request merged**